frontend/src/employee-frontend/components/employee/EmployeeNicknamePage.tsx

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useEffect, useState } from 'react'
+
+import { Loading, Result } from 'lib-common/api'
+import { EmployeeNicknames } from 'lib-common/generated/api-types/pis'
+import Title from 'lib-components/atoms/Title'
+import AsyncButton from 'lib-components/atoms/buttons/AsyncButton'
+import Select from 'lib-components/atoms/dropdowns/Select'
+import Container, { ContentArea } from 'lib-components/layout/Container'
+import { FixedSpaceColumn } from 'lib-components/layout/flex-helpers'
+import { Label, P } from 'lib-components/typography'
+
+import { getPossibleNicknames, setEmployeeNickname } from '../../api/employees'
+import { useTranslation } from '../../state/i18n'
+
+export default React.memo(function EmployeeNicknamePage() {
+  const { i18n } = useTranslation()
+  const [possibleNicknames, setPossibleNicknames] = useState<
+    Result<EmployeeNicknames>
+  >(Loading.of())
+  const [selectedNickname, setSelectedNickname] = useState<string | null>(null)
+
+  useEffect(() => {
+    setPossibleNicknames(Loading.of())
+    void getPossibleNicknames().then(setPossibleNicknames)
+  }, [])
```

Comment:
All of this could be replaced with `useApiState`.


```
const [possibleNicknames, setPossibleNicknames] = useState<
  Result<EmployeeNicknames>
>(Loading.of())

useEffect(() => {
  setPossibleNicknames(Loading.of())
  void getPossibleNicknames().then(setPossibleNicknames)
}, [])
  ```